ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -199,6 +203,71 @@ def batched_step_result_from_proto(
     return BatchedStepResult(obs_list, rewards, done, max_step, agent_id, action_mask)
 
 
+def proto_from_batched_step_result(
+    batched_step_result: BatchedStepResult
+) -> List[AgentInfoProto]:
+    agent_info_protos: List[AgentInfoProto] = []
+    for agent_id in batched_step_result.agent_id:
+        agent_id_index = batched_step_result.agent_id_to_index[agent_id]
+        reward = batched_step_result.reward[agent_id_index]
+        done = batched_step_result.done[agent_id_index]
+        max_step_reached = batched_step_result.max_step[agent_id_index]
+        agent_mask = None
+        if batched_step_result.action_mask is not None:
+            agent_mask = []  # type: ignore
+            for _branch in batched_step_result.action_mask:
+                agent_mask = np.concatenate(
+                    (agent_mask, _branch[agent_id_index, :]), axis=0
+                )
+        observations: List[ObservationProto] = []
+        for all_observations_of_type in batched_step_result.obs:
+            observation = all_observations_of_type[agent_id_index]
+            if len(observation.shape) == 3:
+                observations.append(
+                    ObservationProto(
+                        compressed_data=observation,
```

Comment:
this probably should be float_data - not sure though (@chriselion)

Reply:
It's not safe to assume that (len(shape)==3) => compressed. I need to double-check the BatchedStep result, but I think a PNG observation will have been decompressed to floats already. If that's the case, you should just always do the float_data approach

Reply:
Ok. Then in our 1d test env, should I flatten the faked visual?

Reply:
I think keep it multi-D there, and flatten (I assume that's the reverse of `np.reshape`?) here.